server/src/main/kotlin/org/javacs/kt/codeaction/quickfix/QuickFix.kt

```diff
@@ -16,10 +16,18 @@ interface QuickFix {
 }
 
 fun diagnosticMatch(diagnostic: Diagnostic, range: Range, diagnosticTypes: Set<String>): Boolean =
-    diagnostic.range.equals(range) && diagnosticTypes.contains(diagnostic.code.left)
+    isDiagnosticInRange(diagnostic, range) && diagnosticTypes.contains(diagnostic.code.left)
+
+// for a diagnostic to be in range the lines should be the same, and
+//  the input character range should be within the bounds of the diagnostics range.
+private fun isDiagnosticInRange(diagnostic: Diagnostic, range: Range): Boolean {
+    val diagnosticRange = diagnostic.range
+    return diagnosticRange.start.line == range.start.line && diagnosticRange.end.line == range.end.line &&
+        diagnosticRange.start.character <= range.start.character && diagnosticRange.end.character >= range.end.character
+}
```

Comment:
Maybe it might be worth turning this into an extension function on `Range` (perhaps `contains`, `isSubrange` or a more descriptive name) and moving it into the utils package? Now that I'm thinking about it, we could probably even overload the `in` operator for `Range`, but maybe the method would be more descriptive.

Reply:
That's a great idea! 🙂  Did it in the newest commit. Extension function made the most sense to me, so did that.